src/popup/components/Protocol/EnsureUserSavedMnemonic.tsx

```diff
@@ -77,25 +113,52 @@ export function EnsureUserSavedMnemonic(props: { onComplete: () => void }) {
       <ButtonContainer>
         {buttons.map((button: { word: string; isEnabled: boolean }, index) => (
           <WordButton
+            key={index}
             onClick={() => {
-              const check = mnemonicArr[counter] === button.word;
+              if (button.isEnabled) {
+                setCurrentInput((oldArray) => [...oldArray, button.word]);
+              } else {
+                setCurrentInput(
+                  currentInput.filter((item) => item !== button.word),
+                );
```

Comment:
This would remove all instances of the word from the input if you unclicked one of the buttons.